streamer/streamer.go

```diff
@@ -0,0 +1,102 @@
+package streamer
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+// "Writes can be serialized with respect to other reads and writes. If a read() of file data can be proven (by any means)
+// to occur after a write() of the data, it must reflect that write(), even if the calls are made by different processes"
+// 		- https://pubs.opengroup.org/onlinepubs/009695399/functions/write.html
+// Means we can indeed have many readers concurrently read a file single writer.
+// To avoid readers missing data, mainly around the time the file is closed, our reader just needs to make once
+// last effort to read from the file *after* the file is closed (the writer has exited)
+
+// LiveFileStreamer provides a ReadCloser interface around
+// a file that has at most a single active writer
+// Reads data from the file as the writer writes to it.
+// Returns EOF once the writer closes the file and the last data
+// from the file is read.
+// Closing the provided watcher prematurely also results in graceful termination (with EOF error)
+type LiveFileStreamer struct {
+	// The file to 'LiveStream' from
+	file *os.File
+
+	// A FileWriteWatcher that is watching the associated
+	// File handle
+	writeWatcher *FileWriteWatcher
+
+	// Indicates that the file will receive no more writes
+	writerDone chan struct{}
+
+	// manage close behavior
+	closeOnce *sync.Once
```

Comment:
nit: i don't think this needs to be a pointer, if it's a plain value of type sync.Once there's no need to initialize it.